R/render.R

```diff
@@ -333,6 +337,76 @@ render <- function(input,
     }, add = TRUE)
     env$metadata <- yaml_front_matter
 
+    # hook up output_source to evaluate if specified
```

Comment:
I see -- the idea then would be that the format can do any of the work it needs to in the `pre_knit` hook; does that sound correct? (Ie, we can move all of the code I added to `render` into the format's `pre_knit` hook). I think that should be doable.